e2e/cypress/integration/messaging/message_channel_reference_spec.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Messaging', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18707 - Autocomplete should close if tildes are deleted using backspace', () => {
+        const msg = 'foo';
+
+        // # Make a post
+        cy.postMessage(msg);
+
+        // # Hit the "up" arrow to open the edit modal
+        cy.get('#post_textbox').type('{uparrow}');
+
+        // * Edit modal opens
+        cy.get('#edit_textbox').should('be.visible');
+
+        // # Insert a tilde (~) at the beginning of the post to be edited
+        cy.get('#edit_textbox').type('{ctrl}{home}').type('~');
```

Comment:
That makes sense and in addition, I found that sometimes my test fails because the cursor automatically moves to the end of the post right after opening the edit modal.

So I added `wait(TIMEOUTS.TINY)` before `type('{ctrl}{home}')`.